### PR TITLE
Click element blocks if the result of the click is a open dialog

### PR DESCRIPTION
**Pull request by arbaaz**
Since we can't know the next step until the click returns we are stuck forever waiting for the click to resolve.

### CODE REVIEW

**Comment by arbaaz** [comment]
@rkr090 Please review.

Run below command to test the changes.
`npm run test --testPathPattern confirm`

File: packages/zindi-puppeteer/src/commands.js [hunk 1]
```diff
@@ -131,17 +140,32 @@ module.exports = {
...
+          .click()
+          .then(resolve)
+          .catch(e => (error = e.message));
+      });
+    }
+
     await Promise.all([
+      clickHandler(),
       Promise.race([
+        // below line is added to speedup test run otherwise we have to wait for timeout incase of dialogue.
```
Comment (rkr090) [inline packages/zindi-puppeteer/src/commands.js]: 'dialog' not 'dialogue' :-)

**Comment by rkr090** [comment]
@arbaaz All tests except 'Confirm Dialog' are passing on master ( took 178s ), on this branch tests are taking too much time to execute ( I had to stop it ).

File: packages/zindi-puppeteer/src/commands.js [hunk 2]
```diff
@@ -131,15 +140,27 @@ module.exports = {
...
     let error = null;
+
+    function clickHandler() {
+      return new Promise(resolve => {
+        page.on('dialog', resolve);
+
+        elementHandle
+          .click()
+          .then(resolve)
+          .catch(e => (error = e.message) && resolve());
```
Comment (rkr090) [inline packages/zindi-puppeteer/src/commands.js]: If I reject here, the exception will be handled in index.js in the try/catch block, resolving it here will return the success/error from inside of the 'Click Element' keyword.

Right now we are just disposing of the elementHandle, but we may want to perform other actions later

**Comment by rkr090** [comment]
@arbaaz _a please review the latest commit

**Pull request merged**